Amethyst/Managers/WindowManager.swift

```diff
@@ -237,48 +213,42 @@ private struct ObserveApplicationNotifications {
             .map { _ in }
     }
 
-    private func addObserver(for notification: Notification, application: SIApplication, windowManager: WindowManager) throws {
-        let success = application.observe(notification: notification.string, with: application) { [weak application, weak windowManager] element in
-            guard let application = application, let windowManager = windowManager else {
+    private func addObserver(for notification: Notification, application: AnyApplication<Application>, windowManager: WindowManager<Application>) throws {
+        let success = application.observe(notification: notification.string) { [weak application, weak windowManager] element in
+            guard let application = application, let windowManager = windowManager, let window = element as? Application.Window else {
                 return
             }
 
-            self.handle(notification: notification, element: element, application: application, windowManager: windowManager)
+            self.handle(notification: notification, window: AnyWindow(window), application: application, windowManager: windowManager)
         }
 
         guard success else {
             throw Error.failed
         }
     }
 
-    private func removeObservers(notifications: [Notification], application: SIApplication) {
-        notifications.forEach { application.unobserveNotification($0.string as CFString, with: application) }
+    private func removeObservers(notifications: [Notification], application: AnyApplication<Application>) {
+        notifications.forEach { application.unobserve(notification: $0.string) }
     }
+}
 
-    private func handle(notification: Notification, element: SIAccessibilityElement, application: SIApplication, windowManager: WindowManager) {
+extension ObserveApplicationNotifications {
```

Comment:
Class level comment here would be appreciated, to say exactly what kind of thing we're handling and why